src/v2i-hub/CARMAStreetsPlugin/src/CARMAStreetsPlugin.cpp

```diff
@@ -105,8 +105,17 @@ void CARMAStreetsPlugin::OnConfigChanged(const char *key, const char *value) {
 void CARMAStreetsPlugin::HandleTimeSyncMessage(tmx::messages::TimeSyncMessage &msg, routeable_message &routeableMsg ) {
 	PluginClientClockAware::HandleTimeSyncMessage(msg, routeableMsg);
 	if ( isSimulationMode()) {
-		PLOG(logINFO) << "Handling TimeSync messages!" << std::endl;
-		produce_kafka_msg(msg.to_string(), "time_sync");
+		// TODO: This is a temporary fix for tmx message container property tree
+		// serializing all attributes as strings. This issue needs to be fixed but
+		// is currently out of scope. TMX Messages should be correctly serialize to 
+		// and from json. This temporary fix simply using regex to look for numeric,
+		// null, and bool values and removes the quotations around them.
+		boost::regex exp("\"(null|true|false|[0-9]+(\\.[0-9]+)?)\"");
```

Comment:
I know this is just a fix, but could this potentially cast numeric-only hex values differently than those with letters? If that is how they're passed via json in the existing code at least. Not sure if many, if any, fields will be affected either.

Reply:
Currently this is only being called on a single message, the TimeSyncMessage with two attributes seq and timestep, both are numeric. I do not think we need any protection against hex strings represented in json here.